api.planx.uk/modules/webhooks/service/validateInput/utils.ts

```diff
@@ -0,0 +1,45 @@
+import { isObject } from "lodash";
+import { JSDOM } from "jsdom";
+import createDOMPurify from "dompurify";
+
+// Setup JSDOM and DOMPurify
+const window = new JSDOM("").window;
+const DOMPurify = createDOMPurify(window);
+
+/**
+ * Function which returns -
+ *   - true for valid data
+ *   - false for invalid data
+ */
+type Validator = (input: unknown) => boolean;
+
+/**
+ * Recursively iterate over an object, checking is validator callback condition is met
+ * Fails fast - will return false once first invalid value is encountered
+ */
+export const isObjectValid = (
```

Comment:
I was initially hoping to just use some lodash method to get all nested values from an object so I didn't have to write this (or tests for it!). I don't think there is a method for this however - a few requests on the lodash repo suggest using [flat](https://www.npmjs.com/package/flat) but a whole library for this seems a bit excessive.

If there's a simpler library method or approach for this that I'm just missing please let me know 👍

Reply:
This still feels readable to me, agree a whole library feels like overkill at this stage!